pkg/stringutils/stringutils.go

```diff
@@ -32,6 +32,18 @@ func GenerateRandomASCIIString(n int) string {
 	return string(res)
 }
 
+// Ellipsis truncates a string to fit within maxlen, and appends ellipsis (...).
+// For maxlen of 3 and lower, no ellipsis is appended.
+func Ellipsis(s string, maxlen int) string {
+	if len(s) <= maxlen {
+		return s
+	}
+	if maxlen <= 3 {
+		return s[:maxlen]
+	}
+	return s[:maxlen-3] + "..."
```

Comment:
My personal preference would be not to use the UTF8 ellipsis. I don't really have a good reason to back that up. It just always looks weird when I see three dots crammed into a single position in a monospaced font, and it sometimes causes issues when I copy/paste into something that doesn't handle UTF8 correctly.